aws/rust-runtime/aws-config/src/ecs.rs

```diff
@@ -751,16 +751,22 @@ mod test {
         }
     }
 
-    impl DnsResolver for TestDns {
-        fn resolve_dns(&self, name: String) -> DnsFuture {
-            DnsFuture::ready(Ok(self.addrs.get(&name).unwrap_or(&self.fallback).clone()))
+    impl ResolveDns for TestDns {
+        fn resolve_dns<'a>(&'a self, name: &'a str) -> DnsFuture<'a>
+        where
+            Self: 'a,
```

Comment:
If we say `&'a self`, then why do we also need the `where` bound? What's the difference between those two bounds?